file/json/aws_lambda/index.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const AWS = require('aws-sdk');
+let s3 = new AWS.S3();
+
+let bucketName = "lambda-function-records";
+let bucketParams = { Bucket: bucketName };
+
+let limitNumberOfFileToUpdateInOneRun = 2;
+
+exports.handler = (request, context, callback) => {
+    update(request.state, request.secrets, callback);
+};
+
+async function update(state, secrets, callback) {
+    let response = initializeResponse(state);
+
+    let modifiedFiles = [];
+
+    await s3.listObjects(bucketParams).promise().then(result => {
+        for (let index = 0; index < result.Contents.length; index++) {
+            // Only JSON files to be processed
+            // We can add some pattern
+            if (!result.Contents[index].Key.endsWith(".json") || Date.parse(result.Contents[index].LastModified) <= Date.parse(state.since)) {
+                continue;
+            }
+
+            modifiedFiles.push(result.Contents[index]);
+
+            // If we want to process limited number of files in one lambda execution, so we should only add those number of files
+            if (modifiedFiles.length === limitNumberOfFileToUpdateInOneRun) {
+                response.hasMore = true;
+                break;
+            }
+        }
+    }).catch(err => {
+        callback(err); // Return when some error occurred while listing objects in bucket
+        console.log("Error in listing bucket", JSON.stringify(err) + "\n");
+    });
+
+    // Sort in ascending order
+    modifiedFiles.sort((a, b) => (a.LastModified > b.LastModified) ? 1 : ((b.LastModified > a.LastModified) ? -1 : 0));
+
+    // Process files one by one
+    for (let index = 0; index < modifiedFiles.length; index++) {
```

Comment:
I was not able to use forEach here because we are using await statement inside it (Code was not working when I was testing it with forEach and await)

Reply:
ditto